src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3436,11 +3439,27 @@ private static void DescribeExpression(TextWriter writer, RegexNode node, string
                 for (int i = 0; i < childCount; i++)
                 {
                     int childDepth = skip ? depth : depth + 1;
-                    DescribeExpression(writer, node.Child(i), prefix, childDepth);
+                    DescribeExpression(writer, node.Child(i), prefix, childDepth, capNames);
                 }
             }
         }
 
+        /// <summary>
+        /// Extracts cap name from node if it has type Capture, string.Empty elsewhere
+        /// </summary>
+        /// <param name="node">node</param>
+        /// <param name="capNames">cap name table</param>
+        /// <returns></returns>
+        private static string ExtractCapName(RegexNode node, Hashtable capNames)
+        {
+            if (node.Type != RegexNode.Capture || capNames == null || capNames.Count == 0)
+            {
+                return string.Empty;
+            }
+
+            return capNames.Keys.OfType<string>().FirstOrDefault(i => (int)capNames[i] == node.M);
```

Comment:
We shouldn't use LINQ here.  This should use the same lookup logic that's used in https://github.com/dotnet/runtime/blob/55fecfbc201dc789d71c0903e04f2a779212c671/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs#L341-L355
That logic could even be refactored out into a helper that both Regex and this then call, e.g. putting it into an internal static on RegexParser or somewhere similar.

Reply:
I'm ok with dropping LINQ but I don't think this logic is the one needed :
* Here is an example of what caps look like with Regex.IsMatch("10-563", "^(?<LINE>[0-9]*)-(?<ENDLINE>[0-9]*)$") when calling GroupNameFromNumber
![image](https://user-images.githubusercontent.com/54767641/148126439-3d06b36d-f72b-427f-a0c1-b2ffcaef9ba7.png)
* Here is what caps look like in my case
![image](https://user-images.githubusercontent.com/54767641/148127018-8f17cc71-fec8-4af1-be6e-a920d0261315.png)

Reply:
This:
```C#
using System.Text.RegularExpressions;

Regex r = new Regex("^(?<LINE>[0-9]*)-(?<ENDLINE>[0-9]*)$");
Console.WriteLine(r.GroupNameFromNumber(0));
Console.WriteLine(r.GroupNameFromNumber(1));
Console.WriteLine(r.GroupNameFromNumber(2));
```
outputs:
```
0
LINE
ENDLINE
```